feature-libs/order/components/order-details/order-details-extended/download-invoices/download-order-invoices-dialog.component.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  Component,
+  ChangeDetectionStrategy,
+  OnInit,
+  OnDestroy,
+  ViewChild,
+  ChangeDetectorRef,
+  AfterViewChecked,
+  inject,
+} from '@angular/core';
+import { Order, OrderOutlets } from '@spartacus/order/root';
+import { InvoicesListComponent } from '@spartacus/pdf-invoices/components';
+import { PDFInvoicesFacade } from '@spartacus/pdf-invoices/root';
+import {
+  ICON_TYPE,
+  FocusConfig,
+  LaunchDialogService,
+} from '@spartacus/storefront';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'cx-download-order-invoices-dialog',
+  templateUrl: './download-order-invoices-dialog.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DownloadOrderInvoicesDialogComponent
+  implements OnInit, OnDestroy, AfterViewChecked
+{
+  protected subscription = new Subscription();
+  @ViewChild(InvoicesListComponent, { static: false })
+  public invoiceComponent: InvoicesListComponent;
+  readonly OrderOutlets = OrderOutlets;
+  invoiceCount: number | undefined = undefined;
+  iconTypes = ICON_TYPE;
+  focusConfig: FocusConfig = {
+    trap: true,
+    block: true,
+    autofocus: true,
+    focusOnEscape: true,
+  };
+  order: Order;
+  protected launchDialogService = inject(LaunchDialogService);
+  protected invoicesFacade = inject(PDFInvoicesFacade);
+  protected cdr = inject(ChangeDetectorRef);
+
+  ngOnInit(): void {
+    this.subscription.add(
+      this.launchDialogService.data$.subscribe((data: Order) => {
+        this.order = data;
+      })
+    );
+  }
+
+  ngAfterViewChecked() {
+    this.cdr.detectChanges();
```

Comment:
[MEDIUM]
Calling `detectChanges()` in `ngAfterViewChecked()` at the first sight it looks like a workaround for problems with change detection, that might have roots somewhere else. I'd be happy to understand those problems and hopefully try to fix them in their roots

Reply:
without this 'this.cdr.detectChanges();',     
if (this.invoiceComponent.pagination !== undefined) {
      this.invoiceCount = this.invoiceComponent.pagination.totalResults;
    }
is not getting detected.And shows just spinner and no message

Reply:
I'd love to investigate it further and find the root causes of the problems with change detection and fix it in the roots. But due to tight schedule of release, we might have to postpone it.